Emby.Server.Implementations/Session/SessionManager.cs

```diff
@@ -508,13 +508,17 @@ private SessionInfo GetSessionInfo(
             ArgumentException.ThrowIfNullOrEmpty(deviceId);
 
             var key = GetSessionKey(appName, deviceId);
-
-            CheckDisposed();
-
-            if (!_activeConnections.TryGetValue(key, out var sessionInfo))
+            SessionInfo newSession = CreateSessionInfo(key, appName, appVersion, deviceId, deviceName, remoteEndPoint, user);
+            SessionInfo existingOrNewSession = _activeConnections.GetOrAdd(key, newSession);
+            SessionInfo sessionInfo;
+            if (ReferenceEquals(newSession, existingOrNewSession))
+            {
+                OnSessionStarted(newSession);
+                sessionInfo = newSession;
+            }
+            else
             {
-                sessionInfo = CreateSession(key, appName, appVersion, deviceId, deviceName, remoteEndPoint, user);
-                _activeConnections[key] = sessionInfo;
+                sessionInfo = existingOrNewSession;
```

Comment:
This does not really need an else branch. It uses existingOrNewSession in both cases.